### PR TITLE
Added eventing reconciler and added a pre-action

### PR DESCRIPTION
**Pull request by mfaizanse**
- Added back eventing reconciler (Reverts PR: https://github.com/kyma-incubator/reconciler/pull/1254).
- Added pre-action to delete NATS statefulset if podManagementPolicy is not `Parallel`.

**Related issue(s):**
- https://github.com/kyma-project/kyma/issues/15046#issuecomment-1400250053

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -7,6 +7,9 @@
 # Istio reconciler
 /pkg/reconciler/instances/istio/ @strekm @werdes72 @dariusztutaj @cnvergence @barchw @videlov @triffer
 
+# Eventing reconciler
+/pkg/reconciler/instances/eventing/ @kyma-project/eventing
```
Comment (mfaizanse) [inline CODEOWNERS:11]: https://github.com/orgs/kyma-incubator/teams/eventing/members

File: CODEOWNERS [hunk 2]
```diff
@@ -7,6 +7,9 @@
 # Istio reconciler
 /pkg/reconciler/instances/istio/ @strekm @werdes72 @dariusztutaj @cnvergence @barchw @videlov @triffer
 
+# Eventing reconciler
+/pkg/reconciler/instances/eventing/ @kyma-incubator/eventing
```
Comment (mfaizanse) [inline CODEOWNERS:11]: https://github.com/orgs/kyma-incubator/teams/eventing/members

**Review by friedrichwilken — APPROVED** [review]
/lgtm 
Added some tiny nits. If you think they are irrelevant, just remove the do-not-merge-label. :)

**Review by friedrichwilken — APPROVED** [review]
/lgtm

**Review by jeremyharisch — APPROVED** [review]

**Pull request merged**